### PR TITLE
Add regdomain support

### PR DESCRIPTION
**Pull request by krish2718**
Add build time and runtime regulatory domain SET and GET support.

### CODE REVIEW

File: nrf70_bm_lib/docs/source/nrf70_bm_lib.rst [hunk 1]
```diff
@@ -122,7 +122,7 @@ Optimizing scan operation
 The nRF70 Series BM library provides a single API to perform a Wi-Fi scan operation.
 The scan operation is optimized to provide a wide range of scan configuration parameters.
 
-Please see `Optimizing scan operation <https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/protocols/wifi/scan_mode/scan_operation.html>`_ for more information.
+Please see `Optimizing scan operation <https://docs.nordicsemi.com/bundle/ncs-2.7.0/page/nrf/protocols/wifi/scan_mode/scan_operation.html>`_ for more information.
```
Comment (ioannisg) [inline nrf70_bm_lib/docs/source/nrf70_bm_lib.rst:125]: @krish2718 is this because the information is not part of the latest docs? (Or has moved?)

File: nrf70_bm_lib/include/nrf70_bm_lib.h [hunk 2]
```diff
@@ -249,7 +278,7 @@ typedef void (*nrf70_scan_result_cb_t)(struct nrf70_scan_result *entry);
  * @retval 0 If the operation was successful.
  * @retval -1 If the operation failed.
  */
-int nrf70_bm_init(uint8_t *mac_addr);
+int nrf70_bm_init(uint8_t *mac_addr, struct nrf70_regulatory_info *reg_info);
```
Comment (ioannisg) [inline nrf70_bm_lib/include/nrf70_bm_lib.h:281]: This changes the API. We 'd have to update developers (E.g. release notes) on migration to this release..

File: nrf70_bm_lib/include/nrf70_bm_lib.h [hunk 3]
```diff
@@ -272,6 +301,19 @@ int nrf70_bm_init(uint8_t *mac_addr);
  */
 int nrf70_bm_scan_start(struct nrf70_scan_params *scan_params,
 					 nrf70_scan_result_cb_t cb);
+
+/** @brief Get nRF70 regulatory information.
+ *
+ * This function retrieves the regulatory information of the nRF70 device.
+ * The regulatory information includes the country code, number of channels
```
Comment (ioannisg) [inline nrf70_bm_lib/include/nrf70_bm_lib.h:308]:
```suggestion
 * The regulatory information includes the country code, the number of channels
```

File: nrf70_bm_lib/source/nrf70_bm_core.c [hunk 4]
```diff
@@ -11,8 +11,8 @@
 #include <zephyr/random/random.h>
 #endif /* CONFIG_NRF70_RANDOM_MAC_ADDRESS */
 
-#include "nrf70_bm_core.h"
 #include "nrf70_bm_lib.h"
+#include "nrf70_bm_core.h"
```
Comment (ioannisg) [inline nrf70_bm_lib/source/nrf70_bm_core.c:15]: Was this swap of order intended, @krish2718 ?

File: nrf70_zephyr_shim/source/os/shim.c [hunk 5]
```diff
@@ -26,7 +26,7 @@
 #include "osal_ops.h"
 #include "qspi_if.h"
 
-LOG_MODULE_REGISTER(wifi_nrf, CONFIG_WIFI_NRF700X_SHIM_BUS_LOG_LEVEL);
+LOG_MODULE_REGISTER(wifi_nrf, CONFIG_WIFI_NRF700X_SHIM_LOG_LEVEL);
```
Comment (ioannisg) [inline nrf70_zephyr_shim/source/os/shim.c:29]: I guess this is not related to the reg-domain API extension?

**Review by ioannisg — COMMENTED** [review]
@krish2718 added a few comments (mostly for clarifications, and some for doc improvements)

File: nrf70_bm_lib/include/nrf70_bm_lib.h (hunk 2 above)
Comment (krish2718) [reply nrf70_bm_lib/include/nrf70_bm_lib.h:281]: Sure, but we don't have that concept for this repo, I can add one.

File: nrf70_bm_lib/source/nrf70_bm_core.c (hunk 4 above)
Comment (krish2718) [reply nrf70_bm_lib/source/nrf70_bm_core.c:15]: Yes, the core.h relies on a structure definition in lib.h

File: nrf70_zephyr_shim/source/os/shim.c (hunk 5 above)
Comment (krish2718) [reply nrf70_zephyr_shim/source/os/shim.c:29]: No, general improvement, please see the commit log for details.

Comment (ioannisg) [reply nrf70_zephyr_shim/source/os/shim.c:29]: ok

**Review by ioannisg — APPROVED** [review]
thanks

**Review by sachinthegreen — APPROVED** [review]
